pkg/networkmanager/interfaces.js

```diff
@@ -766,24 +790,15 @@ function NetworkManagerModel() {
     var connections_by_uuid = { };
 
     function set_settings(obj, settings) {
-        var cs = connection_settings(obj);
-        if (cs.uuid)
-            delete connections_by_uuid[cs.uuid];
+        if (obj.Settings && obj.Settings.connection && obj.Settings.connection.uuid)
+            delete connections_by_uuid[obj.Settings.connection.uuid];
```

Comment:
Previously, `connection_settings` was a convenience function that implemented the "look before you leap".  However, as stated now in the comment, there should be a guarantee that `con.Settings.connection` always exists. Because of bugs, this wasn't true, and `connection_settings` allowed us to limp along.  This PR tries to fix the bugs, and now `connection_settings` is a friendly assertion.  We could abort if the guarantee is violated, and maybe we should.  What do you think?

The cases that dealt with incomplete data structures (inside the model), have been rewritten to avoid `connection_settings`, because for them, the guarantee isn't there.